gpu-operator/dra-driver.rst

```diff
@@ -0,0 +1,303 @@
+
+.. literalinclude:: ./manifests/input/imex-channel-injection.yaml
+  :language: yaml
+
+Apply the mainfest.
+
+.. code-block:: console
+
+    $ kubectl apply -f imex-channel-injection.yaml
+
+
+This will create the following resources on your cluster.
+
+- A ComputeDomain resourced named ``imex-channel-injection``.
+
+- A ResourceClaimTemplate named ``imex-channel-0``.
+
+  This is used to request access to a unique IMEX channel on a set of compute nodes.
+  A reference to the ComputeDomain this ResourceClaimTemplate is associated with is passed along as part of a request.
+  Each worker of a multi-node job should reference this ResourceClaimTemplate to ensure that they have access to the same IMEX channel inside the appropriate ComputeDomain
+
+- ResourceClaimTemplate for the DaemonSet.
+
+  This is used to request access to a device that injects ComputeDomain-specific settings for an imex-daemon to run with (config.cfg, nodes_config.cfg, etc.)
+  A reference to the ComputeDomain this ResourceClaimTemplate is associated with is passed along as part of a request.
+  Each worker of the daemonset mentioned below references this ResourceClaimTemplate to ensure they start up IMEX daemons associated with the appropriate ComputeDomain.
+
+- A DaemonSet. 
+
+  This forms one (or more) IMEX domains by running IMEX daemons on the set of nodes in a ComputeDomain that requires them.
+  These daemons “follow” the workload pods to the nodes where they have been scheduled. 
+  Through DRA, these daemons are guaranteed to be fully up and running before the workload pods that triggered their creation are allowed to run.
+
+As the workload ``imex-channel-injection-pod`` referencing the ResourceClaimTemplate for workloads, ``imex-channel-0``, get scheduled, they trigger the DRA Driver to request access to the same IMEX channel on whatever node they land on. 
+
+Once scheduled to a node, the DRA driver adds a Node label for the ComputeDomain to the node where the workload has been scheduled to indicate the node is part of that ComputeDomain.
+This label is used as a NodeSelector on the DaemonSet mentioned above to trigger the scheduling of its pods to specific nodes.
+Once all workloads running in a ComputeDomain have run to completion, the label gets removed (even if the ComputeDomain itself hasn't been deleted yet).
+This allows these nodes to be reused for other ComputeDomains and blocks the workloads from running until the IMEX daemon on that node has been started.
+
+The addition of this Node label triggers the DaemonSet to deploy an IMEX daemon to that node and start running it.
+
+
+Once all daemons have been fully started, the DRA driver unblocks each worker, injects its IMEX channel into the worker and allows it to start running.
+
+
+View the Compute Domain resources on your cluster

```

Comment:
we should also add the MPI nvbandwidth example.

Reply:
Its also described here https://github.com/NVIDIA/k8s-dra-driver-gpu/discussions/249

Reply:
will add that!